lib/main.dart

```diff
@@ -1,115 +1,185 @@
+  void _addTodoItem(String name) {
+    setState(() {
+      _Tasks.add(Task(name: name, checked: false));
+    });
+    _textFieldController.clear();
+  }
 
-  final String title;
+  Future<void> _displayDialog() async {
+    return showDialog<void>(
+      context: context,
+      barrierDismissible: false,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Add task'),
+          content: TextField(
+            controller: _textFieldController,
+            decoration:
+                const InputDecoration(hintText: 'What would you like to do?'),
+          ),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('discontinue'),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+            TextButton(
+              child: const Text('Add'),
+              onPressed: () {
+                Navigator.of(context).pop();
+                _addTodoItem(_textFieldController.text);
+              },
+            )
+          ],
+        );
+      },
+    );
+  }
+}
 
-  @override
-  State<MyHomePage> createState() => _MyHomePageState();
+Widget deleteButton(BuildContext context, todo, String name) {
+  return IconButton(
+    icon: const Icon(Icons.delete_outline),
+    tooltip: "Delete",
+    onPressed: () => showDialog<String>(
+      context: context,
+      builder: (BuildContext context) => AlertDialog(
+        title: const Text('Warning'),
+        content: Text("Do you want to delete '$name'?"),
+        actions: <Widget>[
+          TextButton(
+            onPressed: () => Navigator.pop(context, 'Cancel'),
+            child: const Text('Cancel'),
+          ),
+          TextButton(
+            onPressed: () async {
+              Provider.of<TaskProvider>(context, listen: false)
+                  .removeTask(todo);
+              Navigator.pop(context);
+            },
+            child: const Text('OK'),
+          ),
+        ],
+      ),
+    ),
+  );
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+class TodoItem extends StatelessWidget {
+  TodoItem({
+    required this.todo,
+    required this.onTodoChanged,
+  }) : super(key: ObjectKey(todo));
 
-  void _incrementCounter() {
-    setState(() {
-      // This call to setState tells the Flutter framework that something has
-      // changed in this State, which causes it to rerun the build method below
-      // so that the display can reflect the updated values. If we changed
-      // _counter without calling setState(), then the build method would not be
-      // called again, and so nothing would appear to happen.
-      _counter++;
+  final Task todo;
+  final onTodoChanged;
+
+  TextStyle? _getTextStyle(bool checked) {
+    if (!checked) return null;
+
+    return const TextStyle(
+      color: Color.fromARGB(115, 0, 0, 0),
+      decoration: TextDecoration.lineThrough,
+    );
+  }
```

Comment:
Går att skriva mer kompakt om man vill, även då flytta inline om man så vill.
```suggestion
  TextStyle? _getTextStyle(bool checked) => checked
      ? const TextStyle(
          color: Color.fromARGB(115, 0, 0, 0),
          decoration: TextDecoration.lineThrough,
        )
```